RestaurantReservation.Db/Repositories/RestaurantRepository.cs

```diff
@@ -54,5 +54,15 @@ public async Task DeleteRestaurantAsync(int restaurantId)
                 await _context.SaveChangesAsync();
             }
         }
+
+        public async Task<decimal> CalculateRestaurantTotalRevenueAsync(int restaurantId)
+        {
+            var result = await _context.Restaurants
+                .FromSqlInterpolated($"SELECT dbo.CalculateRestaurantTotalRevenue({restaurantId}) AS TotalRevenue")
```

Comment:
Should we Use EF core and migration?
I mean the requirement is to be familiar with them not to have raw sql

Reply:
Yes, It's a requirement:

> Create a database function to **`Calculate the total revenue generated by a specific restaurant`** using Entity Framework Core migrations.
> Implement the function call in your **`RestaurantReservation.Db`** project, making it accessible from your application.